File: packages/android_intent_plus/lib/src/put_classes/put_float_array.dart
import 'package:android_intent_plus/src/put_classes/base/put_base.dart';

class PutFloatArray extends PutBase<List<double>> {
  PutFloatArray({required String key, required List<double> value})
      : super(key: key, value: value);

  @override
  String get javaClass => 'PutFloatArray';

  factory PutFloatArray.fromJson(
      {required String key, required dynamic value}) {
    return PutFloatArray(key: key, value: List<double>.from(value));
  }

  @override
  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      'key': key,
      'javaClass': javaClass,
      'value': value,
    };
  }
}

File: packages/android_intent_plus/android/src/main/java/dev/fluttercommunity/plus/androidintent/Bundle/PutClasses/PutFloatArray.java
package dev.fluttercommunity.plus.androidintent.Bundle.PutClasses;

import android.os.Bundle;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

import dev.fluttercommunity.plus.androidintent.Bundle.Constants;
import dev.fluttercommunity.plus.androidintent.Bundle.Helpers;
import dev.fluttercommunity.plus.androidintent.Bundle.PutClasses.base.PutBase;

public class PutFloatArray extends PutBase<List<Float>> {

  public static final String JAVA_CLASS = "PutFloatArray";

  public PutFloatArray(String key, List<Float> value) {
    super(key, JAVA_CLASS, value);
  }

  public static void convert(Bundle bundle, PutFloatArray putFloatArray) {
    bundle.putFloatArray(putFloatArray.key, toPrimitiveArray(putFloatArray.value));
  }

  private static float[] toPrimitiveArray(final List<Float> doubleList) {
    final float[] primitives = new float[doubleList.size()];
    for (int i = 0; i < doubleList.size(); i++) {
      primitives[i] = doubleList.get(i);
    }
    return primitives;
  }

  public static PutFloatArray fromJson(JSONObject jsonObject) throws JSONException {
    final ArrayList<Float> arrayList = new ArrayList<>();
    final JSONArray jsonArray = jsonObject.getJSONArray(Constants.VALUE);
    for (int i = 0; i < jsonArray.length(); i++) {
      arrayList.add((float) jsonArray.getDouble(i));
    }
    return new PutFloatArray(jsonObject.getString(Constants.KEY), arrayList);
  }

  @Override
  public JSONObject toJson() throws JSONException {
    JSONObject jsonObject = new JSONObject();
    jsonObject.put("key", key);
    jsonObject.put("javaClass", JAVA_CLASS);
    jsonObject.put("value", new JSONArray(Helpers.convertToArrayList(value)));
    return jsonObject;
  }
}
